src/scripts/main.js

```diff
@@ -1,3 +1,320 @@
+        <option value="Singapore">Singapore</option>
+        <option value="London">London</option>
+        <option value="New York">New York</option>
+        <option value="Edinburgh">Edinburgh</option>
+        <option value="San Francisco">San Francisco</option>
+      </select>
+    </label>
+    <label>
+      Age:
+      <input
+        name="age"
+        type="number"
+        data-qa="age"
+        required
+        min="18"
+        max="90"
+      >
+    </label>
+    <label>
+      Salary:
+      <input
+        name="salary"
+        type="number"
+        min = "0"
+        data-qa="salary"
+        required
+      >
+    </label>
+    <button type="submit"> Save to table </button>
+  </form>
+`);
+
+const form = document.forms[0];
+const officeList = form.querySelector('[data-qa="office"]');
+
+function createRow() {
+  const nameCell = form.name.value;
+  const positionCell = form.position.value;
+  const officeCell = officeList.value;
+  const ageCell = form.age.value;
+  const salary = parseFloat(form.salary.value);
+  const salaryCell = setSalaryValue(salary);
+
+  tableBody.insertAdjacentHTML('beforeend', `
+  <tr>
+    <td>${nameCell}</td>
+    <td>${positionCell}</td>
+    <td>${officeCell}</td>
+    <td>${ageCell}</td>
+    <td>${salaryCell}</td>
+  </tr>
+  `);
+};
+
+function showNotification(type, title, text) {
+  form.insertAdjacentHTML('afterend', `
+    <div class="notification" data-qa="notification">
+      <h1 class="title">${title.toUpperCase()}</h1>
+      <p>${text}</p>
+    </div>
+  `);
+
+  const notification = document.querySelector('.notification');
+
+  notification.classList.add(type);
+
+  setTimeout(() => {
+    notification.remove();
+  }, 3000);
+}
+
+form.name.addEventListener('change', e => {
+  if (!form.name.validity.valid || !form.name.value.trim()) {
+    showNotification(
+      'error',
+      'Incorrect name',
+      'Name length should be at least 4 letters'
+    );
+  }
+});
+
+form.age.addEventListener('change', () => {
+  if (!form.age.validity.valid) {
+    showNotification(
+      'error',
+      'Enter a correct age',
+      'Your age must be from 18 to 90 years'
+    );
+  }
+});
+
+form.salary.addEventListener('change', e => {
+  if (!form.salary.validity.valid) {
+    showNotification(
+      'error',
+      'Enter a correct salary',
+      'Your salary must be greater than or equal to 0'
+    );
+  }
+});
+
+form.addEventListener('submit', (e) => {
+  e.preventDefault();
+
+  if (form.name.value.trim()) {
+    createRow();
+
+    showNotification(
+      'success',
+      'Successful!',
+      'New employee is successfully added to the table'
+    );
+  }
+
+  e.target.reset();
+});
+
+const minNameLength = 4;
+const minAgeLength = 18;
+const maxAgeLegth = 90;
+
+const validScheme = {
+  name: item => item.length >= minNameLength,
+  age: item => +item >= minAgeLength && +item <= maxAgeLegth,
+  salary: item => +item > 0,
+};
+
+tableBody.addEventListener('dblclick', e => {
```

Comment:
Consider simplifying the event listener by creating a separate named function for handling the 'dblclick' event.